lib/nerves_hub_link/device_channel.ex

```diff
@@ -69,8 +84,8 @@ defmodule NervesHubLink.DeviceChannel do
     case Channel.join(socket, topic, params) do
       {:ok, reply, channel} ->
         NervesHubLink.Connection.connected()
-        state = %{state | channel: channel, connected?: true}
-        {:noreply, maybe_update_firmware(reply, state)}
+        _ = handle_join_reply(reply)
```

Comment:
This is minor, but I think we could skip defining this new function here and use `UpdateManager.apply_update(reply)` directly since it utilizes the same logic.

Unless this is predatory to changing NervesHub join reply?